sesman/verify_user_pam.c

```diff
@@ -82,6 +83,56 @@ verify_pam_conv(int num_msg, const struct pam_message **msg,
 }
 
 /******************************************************************************/
+static int DEFAULT_CC
+chauth_pam_conv(int num_msg, const struct pam_message **msg,
+                struct pam_response **resp, void *appdata_ptr)
+{
+    int i;
+    struct pam_response *reply;
+    struct t_user_pass *user_pass;
+
+    reply = (struct pam_response *)
+                g_malloc(sizeof(struct pam_response) * num_msg, 1);
+
+    for (i = 0; i < num_msg; i++)
+    {
+        switch (msg[i]->msg_style)
+        {
+            case PAM_PROMPT_ECHO_ON: /* username */
+                user_pass = (struct t_user_pass *) appdata_ptr;
+                reply[i].resp = g_strdup(user_pass->user);
+                reply[i].resp_retcode = PAM_SUCCESS;
+                break;
+            case PAM_PROMPT_ECHO_OFF: /* password */
+                user_pass = (struct t_user_pass *) appdata_ptr;
+                /* only prompt for old password starts with '('
+                   old pass:        "(current) UNIX password:"
+                   new pass:        "New password:"
+                   retype new pass: "Retype new password:" */
+                if (*(msg[i]->msg) == '(')
```

Comment:
I wish we had a better approach here. Non-English locales can use completely different prompts.